maxscale_exporter.go

```diff
@@ -327,9 +343,88 @@ func (m *MaxScale) parseServers(ch chan<- prometheus.Metric) error {
 		ch <- prometheus.MustNewConstMetric(
 			upMetric.Desc,
 			upMetric.ValueType,
-			serverUp(server.Status),
+			checkStatus(status, SERVER_RUNNING),
 			server.Server, server.Address,
 		)
+
+		statusMetric := m.serverMetrics["server_status"]
+		ch <- prometheus.MustNewConstMetric(
+			statusMetric.Desc,
+			statusMetric.ValueType,
+			checkStatus(status, SERVER_MAINT),
+			server.Server, server.Address, "maintenance",
```

Comment:
I don't see the point in creating a new `server_status` metric as you are using the same value (is the server up or down?) in the `server_up` metric. I understand that you want to be able to alert on all these different status messages but that would still be possible if you attached the status label onto the `server_up` metric.

So that you would get these types of metrics:
`server_up{server="test", address="127.0.0.1", status="running"} 1`
or
`server_up{server="test", address="127.0.0.1", status="master"} 1`

But seeing as a server can have more than one status at a time (e.g. `master` and `running`) it would make sense to attach them to the `status` label in a comma-seperated way. Like:
`server_up{server="test", address="127.0.0.1", status="master,running"} 1`

This way you could see the exact status of a server by looking at the `server_up` metric and also write alerts by targeting specific status messages like so:
`server_up{status=~"*maintenance*"} == 0`